dll/win32/browseui/shellbrowser.cpp

```diff
@@ -806,14 +806,22 @@ HRESULT CShellBrowser::BrowseToPIDL(LPCITEMIDLIST pidl, long flags)
     CComPtr<IShellFolder>                   newFolder;
     FOLDERSETTINGS                          newFolderSettings;
     HRESULT                                 hResult;
+    CLSID                                   clsid;
+    BOOL                                    HasIconViewType;
 
     // called by shell view to browse to new folder
     // also called by explorer band to navigate to new folder
     hResult = SHBindToFolder(pidl, &newFolder);
     if (FAILED_UNEXPECTEDLY(hResult))
         return hResult;
+    IUnknown_GetClassID(newFolder, &clsid);
+    HasIconViewType = clsid == CLSID_MyComputer || clsid == CLSID_ControlPanel ||
```

Comment:
Personally I would have left out CLSID_MyComputer from this list, but since I don't want that PR to be delayed by a big discussion about that tiny detail, I added this only as a comment. I consider this PR to be a giant leap for our filebrowsers usability, although we all know that this will be just an interim solution until we will have the shellbags implemented properly. If you also think that `CLSID_MyComputer` should be excluded here with the aim to also make that use the details-view, please leave me a thumbs up.

Reply:
If anything, I thought that ```CLSID_ShellDesktop``` would be up for debate.